cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -70,9 +73,22 @@ func ValidateMasterConfiguration(c *kubeadm.MasterConfiguration) field.ErrorList
 	allErrs = append(allErrs, ValidateToken(c.Token, field.NewPath("token"))...)
 	allErrs = append(allErrs, ValidateFeatureGates(c.FeatureGates, field.NewPath("feature-gates"))...)
 	allErrs = append(allErrs, ValidateAPIEndpoint(c, field.NewPath("api-endpoint"))...)
+	allErrs = append(allErrs, ValidateProxy(c, field.NewPath("kube-proxy"))...)
 	return allErrs
 }
 
+func ValidateProxy(c *kubeadm.MasterConfiguration, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// Convert to the internal version
+	internalcfg := kubeproxyconfig.KubeProxyConfiguration{}
```

Comment:
Here is the issue!!!! This MUST BE a pointer: `internalcfg := &kubeproxyconfig.KubeProxyConfiguration{}`